File: RabbitMQ/demoPy/Consumer.py
import pika

def main():
    # conectar con RabbitMQ
    connection = pika.BlockingConnection(pika.ConnectionParameters('localhost'))
    channel = connection.channel()

    # Declarar la cola
    channel.queue_declare(queue='hello')

    # Definir una función de callback
    def callback(ch, method, properties, body):
        print(f" [x] Received {body}")

    # Suscribirse a la cola
    channel.basic_consume(queue='hello', on_message_callback=callback, auto_ack=True)

    print(' [*] Waiting for messages. To exit press CTRL+C')
    channel.start_consuming()

if __name__ == '__main__':
    main()

File: RabbitMQ/demo/src/main/java/com/example/Consumer.java
package com.example;

import com.rabbitmq.client.Channel;
import com.rabbitmq.client.Connection;
import com.rabbitmq.client.ConnectionFactory;
import com.rabbitmq.client.DeliverCallback;

public class Consumer {

    public static void main(String[] args) {

        ConnectionFactory factory = new ConnectionFactory();
        factory.setHost("localhost");

        try
        {
            Connection connection = factory.newConnection();
            Channel channel = connection.createChannel();
            channel.queueDeclare("hello", false, false, false, null);

            DeliverCallback deliverCallback = (consumerTag, delivery) -> {
                String message = new String(delivery.getBody(), "UTF-8");
                System.err.println(" [x] Received '" + message + "'");

            };

            channel.basicConsume("hello", true, deliverCallback, consumerTag -> { });

            System.out.println(" [*] Waiting for messages. To exit press Ctrl+C");
            

        }catch(Exception e)
        {
            e.printStackTrace();
        }
    }  
}
